packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/dataStreamSearch/workspaceSelector/workspaceSelector.tsx

```diff
@@ -16,16 +18,34 @@ export interface WorkspaceSelectorProps {
   control: Control<SearchFields>;
   client: IoTTwinMakerClient;
   OnGettingError: (isError: boolean) => void;
+  setValue: UseFormSetValue<SearchFields>;
 }
 
-export const WorkspaceSelector = ({ control, client, OnGettingError }: WorkspaceSelectorProps) => {
+export const WorkspaceSelector = ({ control, client, OnGettingError, setValue }: WorkspaceSelectorProps) => {
   const { workspaces, status } = useWorkspaces({ client });
   const workspaceOptions = createWorkspaceOptions(workspaces);
 
+  const [selectedWorkspace, setSelectedWorkspace] = useLocalStorage<OptionDefinition | null>('selectedWorkspace', null);
+  if (selectedWorkspace) setValue('workspace', selectedWorkspace);
+
   useEffect(() => {
     OnGettingError(status === 'error');
   }, [status, OnGettingError]);
 
+  useEffect(() => {
+    if (workspaceOptions.length) {
+      if (!selectedWorkspace) setSelectedWorkspace(workspaceOptions.at(0));
```

Comment:
I am a bit confused on this logic , 

if (No workspace) --> i.e data still fetching , set ""
if(no localstorage workspace or localstorage workspace not in workspaceslist ) --> set the first workspace.. 

1. i think the "selected" keywork is a bit confusing, i would change it to stored or something... 

Also, the line 37, dont you have to set the value as well ?

Reply:
We have modified the code to be more readable now, and simplifies the condition. 
We don't need setValue again since we update the storedWorkspace and setValue will be handled by line 29.